torchbenchmark/models/soft_actor_critic/__init__.py

```diff
@@ -0,0 +1,243 @@
+        if self.args.self_regularized:
+            # the critic target improvement ratio is annealed during training
+            self.critic_target_imp_slope = (
+                self.args.sr_critic_target_improvement_final - self.args.sr_critic_target_improvement_init
+            ) / self.args.num_steps
+            self.current_target_imp = lambda step: min(
+                self.args.sr_critic_target_improvement_init + self.critic_target_imp_slope * step,
+                self.args.sr_critic_target_improvement_final,
+            )
+
+    def get_module(self):
+        model = self.agent.actor
+        state = self.train_env.reset()
+        action = self.agent.sample_action(state)
+        next_state, reward, done, info = self.train_env.step(action)
+        self.buffer.push(state, action, reward, next_state, done)
+        batch = self.buffer.sample(self.args.batch_size)
+        state_batch, action_batch, reward_batch, next_state_batch, done_batch = batch
+        state_batch = state_batch.to(self.device)
+        return model, (state_batch, )
+        
+    def train(self, niter=1):
+        # Setup
+        self.target_agent.train()
+        done = True
+        for step in range(niter):
+            if done:
+                state = self.train_env.reset()
+                steps_this_ep = 0
+                done = False
+            action = self.agent.sample_action(state)
+            next_state, reward, done, info = self.train_env.step(action)
+            self.buffer.push(state, action, reward, next_state, done)
+            state = next_state
+            steps_this_ep += 1
+            if steps_this_ep >= self.args.max_episode_steps:
+                done = True
+            for _ in range(self.args.gradient_updates_per_step):
+                learn_standard(
+                    buffer=self.buffer,
+                    target_agent=self.target_agent,
+                    agent=self.agent,
+                    actor_optimizer=self.actor_optimizer,
+                    critic_optimizer=self.critic_optimizer,
+                    log_alpha=self.log_alpha,
+                    log_alpha_optimizer=self.log_alpha_optimizer,
+                    target_entropy=self.target_entropy,
+                    batch_size=self.args.batch_size,
+                    gamma=self.args.gamma,
+                    critic_clip=self.args.critic_clip,
+                    actor_clip=self.args.actor_clip,
+                    update_policy=step % self.args.actor_delay == 0,
+                    device=self.device
+                )
+
+            # move target model towards training model
+            if not self.args.self_regularized and (step % self.args.target_delay == 0):
+                soft_update(self.target_agent.critic1, self.agent.critic1, self.args.tau)
+                soft_update(self.target_agent.critic2, self.agent.critic2, self.args.tau)
+    # Set default ninter to 5 because latency of one iteration is too small (~6ms)
```

Comment:
I remember Nikita mentioned 1~5ms is too small as a measure result because it is close to kernel scheduling jiffies (we don't have this problem on CI machine, but other people may want to run TorchBench locally). Also, I hope to avoid the problem like (https://github.com/pytorch/pytorch/issues/56419) where we are too sensitive to a small regression in the code path.

Reply:
If the solution to fast tests is to loop them more, 

and given that we already have 'smart' stuff set up in pytest to determine how many times to repeat each test,

I would suggest just setting the minimum time per test in pytest settings to something you're happy with, rather than adding another place where you might tweak it.

Reply:
I see, I will set the number of iterations to be the minimum that can cover all the code in train().